content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/amp.html

```diff
@@ -0,0 +1,41 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2019 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.carousel="com.adobe.cq.wcm.core.components.models.Carousel"
+  data-sly-use.templates="core/wcm/components/commons/v1/templates.html"
+  class="cmp-carousel"
+  role="group"
+  aria-label="${carousel.accessibilityLabel}"
+  data-cmp-is="carousel">
+    <amp-carousel data-sly-test="${carousel.items && carousel.items.size > 0}"
+      class="cmp-carousel__content"
+      type="slides"
+      width="350"
```

Comment:
@gaetanmarmasse - how did you decide on this default width 350 and height 200 for the carousel? It seems a little opinionated, and can lead to strange renderings given that a carousel can have different components as items with varying content. As far as I can tell there isn't a general adaptive height solution in the `amp-carousel`.

Reply:
@richardhand agreed.
`amp-carousel` requires the `height` and `width` attributes to be set on the element. This is something we've struggled with a little on the frontend side and I believe it is a common challenge with amp around displaying images or elements without known dimensions.
We are following this recommendation: https://amp.dev/documentation/examples/style-layout/how_to_support_images_with_unknown_dimensions/
That said, I agree that this seems very opinionated and misleading. Because we are relying on CSS anyway for the slides within a carousel, the only thing I can think of is to make the `amp-carousel` attributes as followed: `height="1" width="1"`.
This would still work but may not prevent the flicker on the page if the carousel component is high / above the fold...
So basically, these sizes (350 x 200) were our best guess to make sure the element occupies the space it needs and avoid the flicker but I'm open to suggestions. ;)
WDYT?